addons/bus/static/src/js/longpolling_bus.js

```diff
@@ -0,0 +1,252 @@
+odoo.define('bus.Longpolling', function (require) {
+"use strict";
+
+var Bus = require('web.Bus');
+var ServicesMixin = require('web.ServicesMixin');
+
+
+/**
+ * Event Longpolling bus used to bind events on the server long polling return
+ *
+ * trigger:
+ * - window_focus : when the window is focused
+ * - notification : when a notification is receive from the long polling
+ *
+ * @class Longpolling
+ */
+var LongpollingBus = Bus.extend(ServicesMixin, {
+    // constants
+    PARTNERS_PRESENCE_CHECK_PERIOD: 30000,  // don't check presence more than once every 30s
+    ERROR_RETRY_DELAY: 10000, // 10 seconds
+    POLL_ROUTE: '/longpolling/poll',
+
+    // properties
+    _activated: false,
+    _lastNotificationID: 0,
+    _isOdooFocused: true,
+
+    /**
+     * @override
+     */
+    init: function (parent, params) {
+        this._super.apply(this, arguments);
+        this._id = _.uniqueId('bus');
+        this._options = {};
+        this._channels = [];
+
+        // bus presence
+        this._lastPresenceTime = new Date().getTime();
+        this._lastPartnersPresenceCheck = this._lastPresenceTime;
+        $(window).on("focus." + this._id, this._onFocusChange.bind(this, {focus: true}));
+        $(window).on("blur." + this._id, this._onFocusChange.bind(this, {focus: false}));
+        $(window).on("unload." + this._id, this._onFocusChange.bind(this, {focus: false}));
+
+        $(window).on("click." + this._id, this._onPresence.bind(this));
+        $(window).on("keydown." + this._id, this._onPresence.bind(this));
+        $(window).on("keyup." + this._id, this._onPresence.bind(this));
+    },
+    /**
+     * @override
+     */
+    destroy: function () {
+        var self = this;
+        this.stopPolling();
+        $(window).off("focus." + this._id);
+        $(window).off("blur." + this._id);
+        $(window).off("unload." + this._id);
+        _.each('click,keydown,keyup'.split(','), function (evtype) {
```

Comment:
You gain 1 LOC without using `_.each` and removing `var self` on top of the method.